simulator/src/window.rs

```diff
@@ -154,39 +153,41 @@ impl Window {
     {
         self.update(&display);
 
-        loop {
-            let end = self.handle_events();
-            if end {
-                break;
+        'running: loop {
+            if self.events().any(|e| e == SimulatorEvent::Quit) {
+                break 'running;
             }
-
             thread::sleep(Duration::from_millis(20));
         }
     }
 
     /// Handle events
-    pub fn handle_events(&mut self) -> bool {
-        for event in self.event_pump.poll_iter() {
-            match event {
+    /// Return an iterator of all captured SimulatorEvent
+    pub fn events(&mut self) -> impl Iterator<Item = SimulatorEvent> + '_ {
+        let scale = self.scale;
+        let pixel_spacing = self.pixel_spacing;
```

Comment:
Could you move the `map_input_to_point` function into `impl Window` and remove the `scale` and `pixel_spacing` arguments instead?

Reply:
Can move map_input_to_point easily enough, but I need copies to get around accessing self.scale inside of self.event_pump

Reply:
Ok, I didn't noticed that.